libnmstate/nm/ipv4.py

```diff
@@ -77,3 +99,14 @@ def get_info(active_connection):
     info['enabled'] = True
     info['address'] = addresses
     return info
+
+
+def _get_setting_ip_conf(active_connection):
+    """
+    Get NMSettingIP4Config from NMActiveConnection.
+    For any error, return None.
+    """
+    remote_conn = active_connection.get_connection()
```

Comment:
You probably should use https://lazka.github.io/pgi-docs/NM-1.0/classes/ActiveConnection.html#NM.ActiveConnection.get_dhcp4_config

We need to report the *actual* config, not the setting one (which is a desired one from NM perspective).
Could you please confirm that these are the actual and not the setting ones? I'm already confused between the active-connection and the device getters (I assumed it is the same at some point in time)

Reply:
`active_connection.get_dhcp4_config()` is getting the succeeded DHCP information.
For non-going DHCP or failed DHCP, you get `None`. We still need to check profile for that.